pkg/cli/config/adminkubeconfig/new_admin_kubeconfig.go

```diff
@@ -0,0 +1,263 @@
+package adminkubeconfig
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/oc/pkg/cli/config/refreshcabundle"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/cert"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type NewAdminKubeConfigOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+
+	genericclioptions.IOStreams
+}
+
+var (
+	newAdminKubeconfigLong = templates.LongDesc(i18n.T(`
+		Generate, make the server trust, and display a new admin.kubeconfig.
+
+		The key is produced locally and never hits disk.  The public half is pushed to the cluster
```

Comment:
s/never hits disk/is not persisted/  maybe?  "hits disk" is a little colloquial.